### PR TITLE
Theme UI color flexibility, package updates

### PR DESCRIPTION
**Pull request by evanmwillhite**
Changes include:
- Moves colors in Theme UI config to variables for reuse
- Tries to move as much color handling into the config as possible using components like `button` (helps to avoid the hammer solution of having to shadow a component)
- Whatever couldn't be done in config has been moved into new color variables that are less global (e.g., `tabsBg` vs. `accent`)
- Changes the name of "DarkWrapper" because it is only dark in the light theme
- Updates to packages

To Test:
- [ ] Run `yarn` and ensure design matches [the current one](https://emulsify-ds.github.io/gatsby-theme-emulsify-workspace/) with the exception of the title hero which has the background dots removed.

If you run into issues on an existing project, remove the `node_modules` directory and `yarn.lock` and run `yarn` again

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,12 +96,12 @@ You can also show component code in your MDX files using the traditional backtic
 
 #### Other MDX shortcodes available:
 
-`<DarkWrapper>`: Wraps the contents of this component in a "dark" wrapper (opposite of background color). Usage:
+`<ContrastWrapper>`: Wraps the contents of this component in a "dark" wrapper (opposite of background color). Usage:
```
Comment (ModulesUnraveled) [inline README.md]: Do you want to change this part too?
> this component in a "dark" wrapper

File: example/styleguide/0-Getting Started/1-Installation & Usage.md [hunk 2]
```diff
@@ -60,12 +60,12 @@ You can also show component code in your MDX files using the traditional backtic
 
 #### Other MDX shortcodes available:
 
-`<DarkWrapper>`: Wraps the contents of this component in a "dark" wrapper (opposite of background color). Usage:
+`<ContrastWrapper>`: Wraps the contents of this component in a "dark" wrapper (opposite of background color). Usage:
```
Comment (ModulesUnraveled) [inline example/styleguide/0-Getting Started/1-Installation & Usage.md]: Same question about the word "dark" here

File: README.md (hunk 1 above)
Comment (evanmwillhite) [reply README.md]: Yep, done

File: example/styleguide/0-Getting Started/1-Installation & Usage.md (hunk 2 above)
Comment (evanmwillhite) [reply example/styleguide/0-Getting Started/1-Installation & Usage.md]: Yep

**Review by ModulesUnraveled — APPROVED** [review]
Everything seems to work fine, except that the title background still has the dots (you say that it shouldn't in the review notes.)

**Review by ModulesUnraveled — APPROVED** [review]
Everything looks correct now 👍

**Pull request merged**